Gems/AzQtComponentsForPython/Code/CMakeLists.txt

```diff
@@ -0,0 +1,136 @@
+    set_property(TARGET ${AZQTCOMPONENTS_LIBRARY} PROPERTY PREFIX "")
+
+    # Ensure the generated wrapper files have the correct header.
+    set(LICENSE_FILE ${CMAKE_CURRENT_LIST_DIR}/license_header.txt)
+
+    # Set up the options to pass to shiboken.
+    set(shiboken_options --generator-set=shiboken --enable-parent-ctor-heuristic
+        --enable-pyside-extensions --enable-return-value-heuristic --use-isnull-as-nb_nonzero
+        --avoid-protected-hack --language-level=c++17 --debug-level=full
+        --license-file=${LICENSE_FILE}
+        -I${LY_ROOT_FOLDER}/Code/Framework/AzCore
+        -I${LY_ROOT_FOLDER}/Code/Framework/AzQtComponents
+        -I${LY_ROOT_FOLDER}/Code/Framework/AzCore/Platform/${PAL_PLATFORM_NAME} 
+        -I${QT_MODULE_DIR}/include
+        -I${QT_MODULE_DIR}/include/QtWidgets
+        -I${QT_MODULE_DIR}/include/QtCore
+        -I${QT_MODULE_DIR}/include/QtGui
+        -T${PYSIDE2_TYPESYSTEM_PATH}
+        -T${PYSIDE2_SHARE_PATH}
+        --output-directory=${CMAKE_CURRENT_BINARY_DIR}
+    )
+
+    set(generated_sources_dependencies ${wrapped_header} ${typesystem_file})
+
+    add_custom_command(
+        OUTPUT ${WRAPPED_FILES}
+        COMMAND ${SHIBOKEN_GENERATOR_EXE_PATH} ${shiboken_options} ${wrapped_header} ${typesystem_file}
+        DEPENDS ${generated_sources_dependencies} 
+        COMMENT "Running generator for ${typesystem_file}."
+        VERBATIM
+    )
+
+    # Set up the project for building the code generated by shiboken.
+    ly_add_target(
+        NAME ${GEM_NAME}.Editor GEM_MODULE
+        NAMESPACE Gem
+        OUTPUT_NAME ${BINDINGS_LIBRARY}
+        FILES_CMAKE
+            wrapped_files.cmake
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                ${PYSIDE2_INCLUDE_DIR}
+                ${PYTHON_INCLUDE_DIR}
+                ${SHIBOKEN_INCLUDE_DIR}
+                ${PYSIDE2_INCLUDE_DIR}/QtCore
+                ${PYSIDE2_INCLUDE_DIR}/QtGui
+                ${PYSIDE2_INCLUDE_DIR}/QtWidgets
+        BUILD_DEPENDENCIES
+            PUBLIC
+                3rdParty::pyside2
+                Qt5::Widgets
+                Qt5::Core
+                Qt5::Widgets
+                Qt5::Gui
+                ${AZQTCOMPONENTS_LIBRARY}
+                Gem::EditorPythonBindings.Static
+    )
+
+    ly_create_alias(NAME ${MODULE_NAME}.Builders NAMESPACE Gem TARGETS ${MODULE_NAME})
+    ly_create_alias(NAME ${MODULE_NAME}.Tools NAMESPACE Gem TARGETS ${MODULE_NAME})
+
+    target_link_libraries( ${GEM_NAME}.Editor PRIVATE 
+        debug ${PYSIDE2_SHARED_LIBRARIES_D}
+        optimized ${PYSIDE2_SHARED_LIBRARIES}
+    )
+
+    target_link_libraries( ${GEM_NAME}.Editor PRIVATE 
+        debug ${SHIBOKEN_SHARED_LIBRARIES_D}
+        optimized ${SHIBOKEN_SHARED_LIBRARIES}
+    )
+
+    target_sources(${GEM_NAME}.Editor PRIVATE ${WRAPPED_FILES})
+
+    # Building the wrapper files will not work with a unity build.
+    set_target_properties(${GEM_NAME}.Editor PROPERTIES UNITY_BUILD OFF)
```

Comment:
FYI there is a [SKIP_UNITY_BUILD_INCLUSION_FILES](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/azcore_files.cmake#L691-L692) variable that can be set in the `*_files.cmake` to skip wrapper files from being added to the unity build